packages/firestore/src/util/promise.ts

```diff
@@ -23,7 +23,7 @@ export interface Rejecter {
   (reason?: Error): void;
 }
 
-export class Deferred<R> {
+export class Deferred<R = void> {
```

Comment:
What is the purpose of adding `void` to the generic? It seems to compile and run fine even when I removed it.

Reply:
Without it `new Deferred()` is `Deferred<unknown>`. I would like it to default it to `Deferred<void>`.